grails-app/services/com/unifina/service/StreamrClientService.groovy

```diff
@@ -76,7 +77,7 @@ class StreamrClientService {
 					// Create a custom EthereumAuthenticationMethod which doesn't call the API, but instead uses the internal services to
 					// get a sessionToken. Calling the API here can lead to a deadlock situation in some corner cases, because the
 					// service calls "itself" while blocking in a mutex-lock.
-					InternalAuthenticationMethod authenticationMethod = new InternalAuthenticationMethod(nodePrivateKey, ethereumIntegrationKeyService, sessionService)
+					InternalAuthenticationMethod authenticationMethod = new InternalAuthenticationMethod(nodePrivateKey, ethereumIntegrationKeyService, sessionService, SignupMethod.API)
```

Comment:
@hpihkala Is it ok to use SignupMethod.API here? Or should we analyze "Origin" header here, too?

Reply:
@teogeb this is always programmatic, so let's use `API` here.